hcmanager/hcmanager_test_data.go

```diff
@@ -0,0 +1,289 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func getNamespace() string {
+	return "metal3"
+}
+
+func getExtractedHardwareProfile() hwcc.HardwareCharacteristics {
+
+	return hwcc.HardwareCharacteristics{
+		Cpu: &hwcc.Cpu{
+			MaximumCount:    32,
+			MinimumCount:    32,
+			MaximumSpeedMHz: 5000,
+			MinimumSpeedMHz: 3000,
+		},
+		Disk: &hwcc.Disk{
+			MaximumCount:            9,
+			MaximumIndividualSizeGB: 558,
+			MinimumCount:            9,
+			MinimumIndividualSizeGB: 558,
+		},
+		Nic: &hwcc.Nic{
+			MaximumCount: 4,
+			MinimumCount: 4,
+		},
+		Ram: &hwcc.Ram{
+			MaximumSizeGB: 192,
+			MinimumSizeGB: 192,
+		},
+	}
+
+}
+
+func getMissingNicDetails() hwcc.HardwareCharacteristics {
+
+	return hwcc.HardwareCharacteristics{
+		Cpu: &hwcc.Cpu{
+			MaximumCount: 32,
+			MinimumCount: 32,
+		},
+		Disk: &hwcc.Disk{
+			MaximumCount:            9,
+			MaximumIndividualSizeGB: 558,
+			MinimumCount:            9,
+			MinimumIndividualSizeGB: 558,
+		},
+		Ram: &hwcc.Ram{
+			MaximumSizeGB: 192,
+			MinimumSizeGB: 192,
+		},
+	}
+
+}
+
+func getEmptyProfile() hwcc.HardwareCharacteristics {
+	return hwcc.HardwareCharacteristics{}
+}
+
+func getInvalidCPUProfile() hwcc.HardwareCharacteristics {
+
+	return hwcc.HardwareCharacteristics{
+		Cpu: &hwcc.Cpu{
+			MaximumCount:    0,
+			MinimumCount:    0,
+			MaximumSpeedMHz: 0,
+			MinimumSpeedMHz: 0,
+		},
+	}
+}
+
+func getInvalidDiskProfile() hwcc.HardwareCharacteristics {
+
+	return hwcc.HardwareCharacteristics{
+		Disk: &hwcc.Disk{
+			MaximumCount:            0,
+			MaximumIndividualSizeGB: 0,
+			MinimumCount:            0,
+			MinimumIndividualSizeGB: 0,
+		},
+	}
+}
+
+func getInvalidNicProfile() hwcc.HardwareCharacteristics {
+
+	return hwcc.HardwareCharacteristics{
+		Nic: &hwcc.Nic{
+			MaximumCount: 0,
+			MinimumCount: 0,
+		},
+	}
+}
+
+func getInvalidRAMProfile() hwcc.HardwareCharacteristics {
+
+	return hwcc.HardwareCharacteristics{
+		Ram: &hwcc.Ram{
+			MaximumSizeGB: 0,
+			MinimumSizeGB: 0,
+		},
+	}
+}
+
+func getObjectMeta() metav1.ObjectMeta {
+	return metav1.ObjectMeta{Name: "hardwareclassification-test"}
+}
+
+func getExtractedHardwareProfileRuntime() []runtime.Object {
+	expectedHardwareClassification := hwcc.HardwareCharacteristics{
+		Cpu: &hwcc.Cpu{
+			MaximumCount:    1,
+			MinimumCount:    1,
+			MaximumSpeedMHz: 5000,
+			MinimumSpeedMHz: 3000,
+		},
+		Disk: &hwcc.Disk{
+			MaximumCount:            1,
+			MaximumIndividualSizeGB: 500,
+			MinimumCount:            1,
+			MinimumIndividualSizeGB: 500,
+		},
+		Nic: &hwcc.Nic{
+			MaximumCount: 1,
+			MinimumCount: 2,
+		},
+		Ram: &hwcc.Ram{
+			MaximumSizeGB: 8,
+			MinimumSizeGB: 8,
+		},
+	}
+
+	expectedHardwareConfiguration := hwcc.HardwareClassification{
+		Spec: hwcc.HardwareClassificationSpec{
+			HardwareCharacteristics: expectedHardwareClassification,
+		},
+	}
+
+	return []runtime.Object{&expectedHardwareConfiguration}
+}
+
+func getHosts() []runtime.Object {
```

Comment:
there is a lot of repetition in this function. you could  build that list by iterating over one standard element, and then modify what needs to be modified to make each entry as needed

Reply:
done. @maelk @digambar15 kindly review